arangod/IResearch/IResearchViewDBServer.cpp

```diff
@@ -289,80 +321,79 @@ std::function<void()> IResearchViewDBServer::emplace(
     return nullptr;
   }
 
-  irs::string_ref view_id(begin, end - begin);
-  irs::string_ref view_name(end + 1, name.c_str() + name.size() - (end + 1)); // +1 for '_'
-  IResearchViewMeta meta;
-  std::string error;
+  irs::string_ref viewId(begin, end - begin);
+
+  begin = end + 1; // +1 for '_'
+  size -= viewId.size() + 1; // +1 for '_'
 
-  if (!meta.init(info.get(PROPERTIES_FIELD), error)
-      || 1 != meta._collections.size()) {
+  irs::string_ref viewName(begin, size);
+  char* suffix;
+  auto cid = std::strtoull(collectionId.c_str(), &suffix, 10); // 10 for base-10
+
+  if (suffix != collectionId.c_str() + collectionId.size()) {
     LOG_TOPIC(WARN, arangodb::iresearch::TOPIC)
-      << "failed to determine collection ID while constructing IResearch View in database '" << vocbase.name() << "'";
+      << "failed to parse collection ID while constructing IResearch View in database '" << vocbase.name() << "'";
 
     return nullptr;
   }
 
-  auto cid = *(meta._collections.begin());
-  auto wiew = vocbase.lookupView(view_id);
+  auto wiew = vocbase.lookupView(viewId); // always look up by view ID since it cannot change
 
   // create DBServer view
   if (!wiew) {
+    static const std::function<bool(irs::string_ref const& key)> acceptor = [](
+        irs::string_ref const& key
+    )->bool {
+      return key != ID_FIELD && key != IS_SYSTEM_FIELD && key != NAME_FIELD; // ignored fields
+    };
     arangodb::velocypack::Builder builder;
 
     builder.openObject();
-    builder.add(
+    builder.add(ID_FIELD, arangodb::velocypack::Value(viewId));
+    builder.add(NAME_FIELD, arangodb::velocypack::Value(viewName));
+    builder.add(// mark the view definition as a DBServer instance
       VIEW_CONTAINER_MARKER,
       arangodb::velocypack::Value(arangodb::velocypack::ValueType::Null)
     );
 
-    if (!mergeSlice(builder, info)) {
+    if (!mergeSliceSkipKeys(builder, info, acceptor)) {
       LOG_TOPIC(WARN, arangodb::iresearch::TOPIC)
```

Comment:
Don't we have logic for this in VPackCollection ?

Reply:
IIRC the version in VelocyPackHelpers internally creates a new arangodb::velocypack::Builder instead of simply not adding keys to the existing builder